test/functional/system/blocks/chain/delete_last_block_db_trs.js

```diff
@@ -0,0 +1,403 @@
+		blockAccount1 = randomUtil.account();
+		fundTrsForAccount1 = lisk.transaction.createTransaction(
+			blockAccount1.address,
+			transferAmount,
+			accountFixtures.genesis.password
+		);
+
+		fundTrsForAccount1.senderId = accountFixtures.genesis.address;
+
+		localCommon.createValidBlock(library, [fundTrsForAccount1], (err, b) => {
+			expect(err).to.not.exist;
+			block = b;
+			library.modules.blocks.chain.applyBlock(block, true, done);
+		});
+	});
+
+	describe('popLastBlock', () => {
+		let processExit;
+
+		beforeEach(done => {
+			processExit = sinonSandbox.stub(process, 'exit');
+			return done();
+		});
+
+		afterEach(done => {
+			processExit.restore();
+			done();
+		});
+
+		describe('when it fails', () => {
+			describe('when loadBlockSecondLastBlockStep fails', () => {
+				beforeEach(done => {
+					block.previousBlock = null;
+					library.modules.blocks.lastBlock.set(block);
+					return done();
+				});
+
+				it('should call process.exit', done => {
+					library.modules.blocks.chain.deleteLastBlock(err => {
+						expect(err).to.exist;
+						expect(process.exit.calledOnce).to.equal(true);
+						return done();
+					});
+				});
+			});
+
+			describe('when undoStep fails', () => {
+				let setAccountAndGet;
+				beforeEach(done => {
+					// Artifically fail setAccountAndGet so we can check that test fails
+					setAccountAndGet = library.modules.accounts.setAccountAndGet;
+					sinonSandbox
+						.stub(library.modules.accounts, 'setAccountAndGet')
+						.callThrough()
+						.withArgs({
+							address: fundTrsForAccount1.recipientId,
+						})
+						.callsArgWith(1, 'err');
+					done();
+				});
+
+				afterEach(done => {
+					library.modules.accounts.setAccountAndGet = setAccountAndGet;
+					done();
+				});
+
+				it.skip('should not have perform undoStep on transactions of block', done => {
+					library.modules.blocks.chain.deleteLastBlock(err => {
+						expect(err).to.exist;
+						expect(process.exit.calledOnce).to.equal(true);
+						localCommon
+							.getAccountFromDb(library, fundTrsForAccount1.recipientId)
+							.then(account => {
+								expect(account.mem_accounts.balance).to.equal(
+									transferAmount.toString()
+								);
+								done();
+							});
+					});
+				});
+			});
+
+			describe('when undoUnconfirmStep fails', () => {
+				let merge;
+				beforeEach(done => {
+					// Artifically fail setAccountAndGet so we can check that test fails
+					merge = library.logic.transaction.scope.account.merge;
+
+					sinonSandbox
+						.stub(library.logic.transaction.scope.account, 'merge')
+						.callThrough()
+						.withArgs(fundTrsForAccount1.senderId, {
+							u_balance: fundTrsForAccount1.amount + fundTrsForAccount1.fee,
+						})
+						.callsArgWith(2, 'err');
+					done();
+				});
+
+				afterEach(done => {
+					library.logic.account.merge = merge;
+					done();
+				});
+
+				it('should not have perform undoStep on transactions of block', done => {
```

Comment:
You are not really checking here if undoStep was performed or not. Also, please split this test and the following on 2 parts:
```
it('should call process.exit with 1')
it('should not change balance in mem_accounts table')
```

Reply:
"You are not really checking here if undoStep was performed or not"
I am actuall, I am checking whether balance for mem_account was reverted or not. 
" Also, please split this test and the following on 2 parts:"
Yeah, on it!